### ISSUE TITLE
web: AnimatedComponent raises a deprecation warning for findDOMNode in v3.12.0

### ISSUE THREAD

**Issue by MaxGraey** [software-mansion/react-native-reanimated#6072]
### Description

With latest update, Expo project for web builds start to show error:

```text
Warning: findDOMNode is deprecated and will be removed in the next major release. 
Instead, add a ref directly to the element you want to reference. 
Learn more about using refs safely here: https://reactjs.org/link/strict-mode-find-node
    at AnimatedComponent
    ...
```

### Steps to reproduce

start expo  for web (or eact-native + react-native-for-web) with dev mode enabled.

### Snack or a link to a repository

-

### Reanimated version

3.12.0

### React Native version

0.74

### Platforms

Web

### JavaScript runtime

V8

### Workflow

Expo Dev Client

### Architecture

Fabric (New Architecture)

### Build type

Debug app & dev bundle

### Device

None

### Acknowledgements

Yes

**Comment by MaxGraey** [comment]
It looks like this was introduced in #5960
here: https://github.com/software-mansion/react-native-reanimated/commit/deb782556d03d86bcfdc8e9e92f06a7491aa757f#diff-8ee19000960c96e45bf813296fe1bc00203d9b26bc18e8c7f7a4dae6ca9682d7R238

**Comment by szydlovsky** [comment]
Hey @MaxGraey this `finNodeHandle` is not even needed on Web since the tags are used only for native platforms. The code is refactored in https://github.com/software-mansion/react-native-reanimated/pull/6030 and it should fix the warning.

**Comment by km-tr** [comment]
@szydlovsky 
I'm getting the same error in 3.12.1 as well. What do you think?

**Comment by szydlovsky** [comment]
@km-tr could you please provide me with an example that I can use to reproduce the warning? I have tried to get it in 3.12.1 to no avail.

**Comment by szydlovsky** [comment]
@km-tr Yeah, I interpreted our release notes wrongly 😄 The fix will be present in 3.13 version

**Comment by km-tr** [comment]
@szydlovsky 
I understand. I'll wait for 3.13.

**Comment by suchcodemuchwow** [comment]
I'm still getting the same error in version 3.16.1
Any workarounds?

**Comment by jose920405** [comment]
Same issue in 3.16.1. 

In general, everything is working fine. It is just annoying having this red warning in the console

**Comment by nmpereira** [comment]
The warning still persists in 3.17.1, but everything works as expected. Is there a workaround for this?

Should the issue be open until there's a fix for it?
